File: M80AttributedLabel/OCDemo/Classes/LinksViewController.m
//
//  LinksViewController.m
//  M80AttributedLabel
//
//  Created by amao on 5/21/14.
//  Copyright (c) 2014 www.xiangwangfeng.com. All rights reserved.
//

#import "LinksViewController.h"
#import "M80AttributedLabel.h"


@interface LinksViewController ()<M80AttributedLabelDelegate>

@end

@implementation LinksViewController

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    M80AttributedLabel *label = [[M80AttributedLabel alloc]initWithFrame:CGRectZero];
    
    NSString *text  = @"The game which I current play is hearthstone,and its website is www.hearthstone.com.cn";
    NSRange range   = [text rangeOfString:@"hearthstone"];
    label.text      = text;
    [label addCustomLink:[NSValue valueWithRange:range]
                forRange:range];
    label.delegate = self;
    
    
    label.frame     = CGRectInset(self.view.bounds,20,20);
    
    [self.view addSubview:label];

}

- (void)m80AttributedLabel:(M80AttributedLabel *)label
             clickedOnLink:(id)linkData
{
    NSString *message = [NSString stringWithFormat:@"link data is %@:%@",[[linkData class] description],linkData];
    
    UIAlertController *controller = [UIAlertController alertControllerWithTitle:@"you click a link"
                                                                        message:message
                                                                 preferredStyle:UIAlertControllerStyleAlert];
    UIAlertAction *action = [UIAlertAction actionWithTitle:@"Cancel"
                                                     style:UIAlertActionStyleCancel
                                                   handler:nil];
    [controller addAction:action];
    [self presentViewController:controller
                       animated:YES
                     completion:nil];
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event
{
    NSLog(@"touch view controller");
}
@end

File: SwiftDemo/Classes/LinksViewController.swift
//
//  LinksViewController.swift
//  SwiftDemo
//
//  Created by amao on 2016/10/12.
//  Copyright © 2016年 amao. All rights reserved.
//

import UIKit

class LinksViewController: BaseViewController,M80AttributedLabelDelegate {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.title = "Link"
        
        let text = "The game which I current play is hearthstone,and its website is www.hearthstone.com.cn"
        let label = M80AttributedLabel()
        label.text = text
        
        let range = (text as NSString).range(of: "hearthstone")
        if range.location != NSNotFound {
            label.addCustomLink(NSValue(range:range),for: range)
        }
        label.delegate = self
        
        label.frame = self.view.bounds.insetBy(dx: 20, dy: 20)
        
        self.view.addSubview(label)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    func m80AttributedLabel(_ label: M80AttributedLabel, clickedOnLink linkData: Any) {
        
        let message = "LinkData is \(linkData) "
        let controller = UIAlertController(title: "you click a link", message: message, preferredStyle: .alert)
        let action = UIAlertAction(title: "Cancel", style: .cancel, handler: nil)
        controller.addAction(action)
        self.present(controller, animated: true, completion: nil)
    }

    
}
